app_flutter/lib/status_grid.dart

```diff
@@ -217,23 +127,22 @@ class StatusGrid extends StatelessWidget {
               /// row of [TaskIcon] introduces.
               final int index = gridIndex - columnCount;
               if (index < 0) {
-                return TaskIcon(task: taskIconRow[gridIndex - 1]);
+                return TaskIcon(task: taskMatrix.sampleTask(gridIndex - 1));
               }
 
-              final int statusIndex = index ~/ columnCount;
+              final int row = index ~/ columnCount;
               if (index % columnCount == 0) {
-                return CommitBox(commit: statuses[statusIndex].commit);
+                return CommitBox(commit: statuses[row].commit);
               }
 
-              final int taskIndex = (index % columnCount) - 1;
-              if (taskMatrix[statusIndex][taskIndex] == null) {
+              final int column = (index % columnCount) - 1;
+              final Task task = taskMatrix.task(row, column);
+              if (task == null) {
                 /// [Task] was skipped so don't show anything.
                 return const SizedBox();
               }
 
-              return TaskBox(
-                task: taskMatrix[statusIndex][taskIndex],
-              );
+              return TaskBox(task: task);
```

Comment:
To reduce the magic in this function, we can create another 2D matrix (say `TaskView`) that has the same dimensions as this GridView. And populate `TaskView` with simple for loops to traverse the TaskMatrix, and add CommitBox.

This approach would use more memory, but not bad on time complexity as the TestMatrix needs to be traversed anyway.

Reply:
I plan to refactor `StatusGrid` after v1 parity to move out the `TaskIcon` row and `CommitBox` column from the `GridView` so both of those show no matter where you scroll in the `TaskMatrix`. Such as if you scroll far to the right, you should still see the `CommitBox` column. If you scroll to very old commits, you should still be able to see the `TaskIcon` row.

`TaskView` is a good name for what would be left of the `GridView`, and I will leave a TODO for now for https://github.com/flutter/flutter/issues/43376